rezasm-source/rezasm-core/src/instructions/argument_type.rs

```diff
@@ -14,93 +11,28 @@ pub enum ArgumentType {
     Input(InputTarget),
 }
 
-pub trait Downcast {
-    fn downcast<T: Input>(&self) -> Option<&T>;
-}
-
-impl Downcast for ArgumentType {
-    fn downcast<T: Input>(&self) -> Option<&T> {
+impl Into<InputOutputTarget> for ArgumentType {
```

Comment:
No need to panic, we can implement TryInto<T> and match on the output when we use it.

Reply:
And what happens if the Result of TryInto is an Error? We're still in the same position. If something goes wrong when we're converting from ArgumentType to any type of Target that ought to be irrecoverable. It's something that can only ever happen if there's a bug in the implementation of `Line::new()`

I'm fine making changing this to TryInto, but the result will probably just be panicking anyway unless we change the way the Line constructor in a way that I can't think of at the moment.